File: common/src/main/java/com/terrasect/common/util/MutablePointContext.java
package com.terrasect.common.util;

import net.minecraft.world.level.levelgen.DensityFunction;

public final class MutablePointContext implements DensityFunction.FunctionContext {
    private int x;
    private int y;
    private int z;

    public void set(int x, int y, int z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    @Override
    public int blockX() {
        return x;
    }

    @Override
    public int blockY() {
        return y;
    }

    @Override
    public int blockZ() {
        return z;
    }
}

File: common/src/test/kotlin/terrasect/testing/MutablePointContext.kt
package terrasect.testing

import net.minecraft.world.level.levelgen.DensityFunction

class MutablePointContext : DensityFunction.FunctionContext {
  private var x = 0
  private var y = 0
  private var z = 0

  fun set(x: Int, y: Int, z: Int) {
    this.x = x
    this.y = y
    this.z = z
  }

  override fun blockX() = x
  override fun blockY() = y
  override fun blockZ() = z
}
